core/app/[locale]/(default)/(faceted)/brand/[slug]/page.tsx

```diff
@@ -80,229 +47,185 @@ const createBrandSearchParamsCache = cache(async (props: Props) => {
+  const { brand, settings } = await getBrandPageData(brandId);
 
-  return t('remove');
-}
+  if (!brand) {
+    return notFound();
+  }
 
-async function getMaxCompareLimitMessage(): Promise<string> {
-  const t = await getTranslations('Components.ProductCard.Compare');
+  const productComparisonsEnabled =
+    settings?.storefront.catalog?.productComparisonsEnabled ?? false;
 
-  return t('maxCompareLimit');
-}
+  const streamableFacetedSearch = Streamable.from(async () => {
+    const searchParams = await props.searchParams;
+    const customerAccessToken = await getSessionCustomerAccessToken();
+    const currencyCode = await getPreferredCurrencyCode();
 
-async function getEmptyStateTitle(): Promise<string> {
-  const t = await getTranslations('Faceted.Brand.Empty');
+    const searchParamsCache = await createBrandSearchParamsCache(slug);
+    const parsedSearchParams = searchParamsCache?.parse(searchParams) ?? {};
 
-  return t('title');
-}
+    const search = await fetchFacetedSearch(
+      {
+        ...searchParams,
+        ...parsedSearchParams,
+        brand: [slug],
+      },
+      currencyCode,
+      customerAccessToken,
+    );
 
-async function getEmptyStateSubtitle(): Promise<string> {
-  const t = await getTranslations('Faceted.Brand.Empty');
+    return search;
+  });
 
-  return t('subtitle');
-}
+  const streamableProducts = Streamable.from(async () => {
+    const format = await getFormatter();
+
+    const search = await streamableFacetedSearch;
+    const products = search.products.items;
+
+    return products.map((product) => ({
+      id: product.entityId.toString(),
+      title: product.name,
+      href: product.path,
+      image: product.defaultImage
+        ? { src: product.defaultImage.url, alt: product.defaultImage.altText }
+        : undefined,
+      price: pricesTransformer(product.prices, format),
+      subtitle: product.brand?.name ?? undefined,
+    }));
+  });
 
-async function getFiltersPanelTitle(): Promise<string> {
-  const t = await getTranslations('Faceted.FacetedSearch');
+  const streamableTotalCount = Streamable.from(async () => {
+    const search = await streamableFacetedSearch;
 
-  return t('filters');
-}
+    return search.products.collectionInfo?.totalItems ?? 0;
+  });
 
-async function getRangeFilterApplyLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.FacetedSearch.Range');
+  const streamablePagination = Streamable.from(async () => {
+    const search = await streamableFacetedSearch;
 
-  return t('apply');
-}
+    return pageInfoTransformer(search.products.pageInfo);
+  });
 
-async function getResetFiltersLabel(): Promise<string> {
-  const t = await getTranslations('Faceted.FacetedSearch');
+  const streamableFilters = Streamable.from(async () => {
+    const searchParams = await props.searchParams;
+    const searchParamsCache = await createBrandSearchParamsCache(slug);
+    const parsedSearchParams = searchParamsCache?.parse(searchParams) ?? {};
```

Comment:
Should we extract the parsed search params so we can reuse it both here and [above](https://github.com/bigcommerce/catalyst/blob/8bb203a71b9ea8e304bdb00c3890e5fab4f61ab9/core/app/%5Blocale%5D/(default)/(faceted)/brand/%5Bslug%5D/page.tsx#L101)?